### DISCUSSION TITLE
set_tree

### DISCUSSION THREAD

**Discussion by SrTobi** [i3/i3#5661] [Ideas]
Hi,

I want to tinker with automatically "saving"/"restoring" layouts in i3.
For this I need a method of setting the layout of a complete workspace.
I know there is the `append_layout` command, but it seems to only create the placeholder...
not swallowing existing windows... Am I missing something here?

This would still be a workaround for me.
Ideally I want an ipc command `set_tree` which I give a json similar to `get_tree` (with window ids and so on)
and the command sets the tree accordingly.
I can make a PR for this, but wanted to ask if this exists already in another form (and I just didn't see it) and if this would be welcomed or if there is something speaking against it?

**Comment by mahrud** [comment]
Along the same lines, is there a command similar to `append_layout` that doesn't "append" but rather discards the current layout and forces in the new layout?
